cpp/src/strings/strings_column_factories.cu

```diff
@@ -0,0 +1,122 @@
+            return ( item.first ? (int32_t)item.second : 0 );
+        },
+        thrust::plus<int32_t>() );
+
+    // get number of bytes (last offset value)
+    size_type bytes = thrust::device_pointer_cast(offsets_view.data<int32_t>())[count-1];
+
+    // count nulls
+    size_type null_count = thrust::transform_reduce( execpol->on(stream),
+        thrust::make_counting_iterator<size_type>(0), thrust::make_counting_iterator<size_type>(count),
+        [d_strs] __device__ (size_type idx) { return (size_type)(d_strs[idx].first==nullptr); },
+        0, thrust::plus<size_type>() );
+
+    // build null_mask
+    mask_state state = mask_state::UNINITIALIZED;
+    if( null_count==0 )
+        state = mask_state::UNALLOCATED;
+    else if( null_count==count )
+        state = mask_state::ALL_NULL;
+    auto null_mask = create_null_mask(count, state, stream, mr);
+    if( (null_count > 0) && (null_count < count) )
+    {
+        uint8_t* d_null_mask = static_cast<uint8_t*>(null_mask.data());
+        CUDA_TRY(cudaMemsetAsync(d_null_mask, 0, null_mask.size(), stream));
+        thrust::for_each_n(execpol->on(stream), thrust::make_counting_iterator<size_type>(0), (count/8),
+            [d_strs, count, d_null_mask] __device__(size_type byte_idx) {
+                unsigned char byte = 0; // set one byte per thread -- init to all nulls
+                for( size_type i=0; i < 8; ++i )
+                {
+                    size_type idx = i + (byte_idx*8);  // compute d_strs index
+                    byte = byte >> 1;                  // shift until we are done
+                    if( idx < count )                  // check boundary
+                    {
+                        if( d_strs[idx].first )
+                            byte |= 128;               // string is not null, set high bit
+                    }
+                }
+                d_null_mask[byte_idx] = byte;
+            });
+    }
+
+    // build chars column
+    auto chars_column = make_numeric_column( data_type{INT8}, bytes, mask_state::UNALLOCATED, stream, mr );
+    auto chars_view = chars_column->mutable_view();
+    auto d_chars = chars_view.data<int8_t>();
+    auto d_offsets = offsets_view.data<int32_t>();
+    thrust::for_each_n(execpol->on(stream), thrust::make_counting_iterator<size_type>(0), count,
+          [d_strs, d_offsets, d_chars] __device__(size_type idx){
+              // place individual strings
+              auto item = d_strs[idx];
+              if( item.first )
+              {
+                  size_type offset = (idx ? d_offsets[idx-1] : 0);
+                  memcpy(d_chars + offset, item.first, item.second );
+              }
+          });
+
+    // build children vector
+    std::vector<std::unique_ptr<column>> children;
+    children.emplace_back(std::move(offsets_column));
+    children.emplace_back(std::move(chars_column));
+
+    return std::make_unique<column>(
+        data_type{STRING}, 0, rmm::device_buffer{0,stream,mr},
```

Comment:
We want to have the `column` carry the number of elements it has. Even though a `STRING` column doesn't have any data (it's all in the children), we still want to be able to query the number of elements directly.

Reply:
Hit this one right away
https://github.com/rapidsai/cudf/blob/af668b8500c09ae87177e286eb9d4a6f846fd15b/cpp/src/column/column_view.cpp#L45